### PR TITLE
xbgpu engine test cleanup

### PR DESCRIPTION
**Pull request by bmerry**
Remove a number of warts in xbgpu/test_engine.py. Refer to the individual commits for more details - they might also be easier to follow than the whole PR at once. It's shorter, has less duplication, and runs a bit faster.

Checklist (if not applicable, edit to add `(N/A)` and mark as done):

- [x] (N/A) If dependencies are added/removed: update ``setup.cfg`` and appropriate requirements files
- [x] (N/A) If modules are added/removed: use sphinx-apidoc to update files in ``doc/``
- [x] Ensure copyright notices are present and up-to-date
- [x] (N/A) If qualification tests are changed: attach a sample qualification report
- [x] (N/A) If design has changed: ensure documentation is up to date
- [x] (N/A) If ICD-defined sensors have been added: update `fake_servers.py` in katsdpcontroller to match

Closes NGC-832.

### CODE REVIEW

**Comment by bmerry** [comment]
Not really sure if this needs 2 reviewers, but adding you both in case there are disagreements about style or the way I've used fixtures everywhere.

**Comment by bmerry** [comment]
Whoops, I got confused about which PR I was looking at and accidentally merged this. @james-smith-za @amishatishpatel could one of you do a review even though this is merged, and I'll make a new PR for requested changes?

File: test/xbgpu/test_engine.py [hunk 1]
```diff
@@ -107,26 +140,18 @@ def generate_expected_output(
         output_array[..., 1] = 1
         return output_array
     for b in range(batch_start_idx, batch_start_idx + num_batches):
-        sign = pow(-1, b)
         for c in range(channels):
-            h = np.empty((antennas, 2), np.int32)
-            v = np.empty((antennas, 2), np.int32)
+            in_data = np.empty((antennas, 2, 2), np.int32)
```
Comment (james-smith-za) [inline test/xbgpu/test_engine.py:144]: While you're busy tidying up, I'd suggest using `N_POLS` and `COMPLEX` here.

**Review by james-smith-za — COMMENTED** [review]
Overall I like it, just one comment.

File: test/xbgpu/test_engine.py (hunk 1 above)
Comment (bmerry) [reply test/xbgpu/test_engine.py:144]: Ok, opened #483 for it.

**Pull request merged**